sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java

```diff
@@ -36,18 +42,19 @@ public static void applyMetadata(Context context, SentryOptions options) {
     }
   }
 
-  public static boolean isAutoInit(Context context, SentryOptions options) {
+  public static boolean isAutoInit(Context context, ILogger logger) {
+    if (context == null) throw new IllegalArgumentException("The application context is required.");
+
     boolean autoInit = true;
     try {
       Bundle metadata = getMetadata(context);
       if (metadata != null) {
         autoInit = metadata.getBoolean(AUTO_INIT, true);
-        options.getLogger().log(SentryLevel.DEBUG, "Auto-init: %s", autoInit);
+        logger.log(SentryLevel.DEBUG, "Auto-init: %s", autoInit);
```

Comment:
Do we nave to null check for `logger`? Since this method is public

Reply:
ideally should not be public, but for now yeah, will do NPE check, good catch.